### PR TITLE
Bug 1421643 - Use existing openshift/origin image instead of new openshift/diagnostics-deployer

### PR DESCRIPTION
**Pull request by pravisankar**
Any new image like 'openshift/diagnostics-deployer' incurs build/lifecycle costs to maintian and
diagnostics-deployer image has only small block of shell code. To alleviate this problem,
now the script is embedded into the pod definition and openshift/origin is used as diagnostics deployer
image. On dev machines, currently openshift/origin is close to 800MB but we expect the size to be under 200MB when it is released (compressed, debug headers removed).

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/networking @sdodson PTAL

**Comment by danwinship** [comment]
weird that this even works, but lgtm I guess

**Comment by dcbw** [comment]
lgtm

**Comment by pravisankar** [comment]
@smarterclayton can you please review/merge?

**Comment by smarterclayton** [comment]
Outside the script looks fine.  Lgtm as well [merge] thanks.

**Comment by pravisankar** [comment]
Unrelated errors:
[ERROR] PID 8: hack/build-rpm-release.sh:35: `tito tag --undo --offline` exited with status 1
fatal: [localhost]: FAILED! => {"failed": true, "msg": "could not locate file in lookup: /usr/share/ansible/openshift-ansible-gce/playbooks/files/gce.json"}

re[test]

File: pkg/diagnostics/network/objects.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 const (
-	diagnosticsImage           = "openshift/diagnostics-deployer"
+	diagnosticsImage           = "openshift/origin"
```
Comment (sdodson) [inline pkg/diagnostics/network/objects.go]: Can this be updated to use the image format string so that we don't have to patch OCP?

Comment (pravisankar) [inline pkg/diagnostics/network/objects.go]: I don't see existing image format string for 'openshift/origin', there is openshift/origin-${component}:${version} for system components but that doesn't match with what we need. 
Supporting this as config involves creating Network diags section with an option in master-config 
and this needs to be propagated all the way to GetNetworkDiagnosticsPod(). 
Is it easy to patch OCP instead?

Comment (sdodson) [inline pkg/diagnostics/network/objects.go]: Is it appropriate to use the image format string and strip the hyphen? That's the image name we'd want 'openshift/origin' and 'openshift3/ose'

**Comment by pravisankar** [comment]
Added a new commit that makes network diagnostics pod image configurable (default: openshift/origin)
For OCP, user can use openshift3/origin (any image that has shell support)
@sdodson @dcbw @danwinship PTAL

**Comment by eparis** [comment]
Looks like this fails on verifying completions. Maybe other things...  Please leave a note when you fix the verifications.

**Comment by pravisankar** [comment]
Updated auto generated docs/bash completions

**Comment by pravisankar** [comment]
[test]

**Comment by eparis** [comment]
[merge]

**Comment by eparis** [comment]
[merge]

**Comment by smarterclayton** [comment]
[merge] dirty git repo flake

On Feb 21, 2017, at 7:45 PM, OpenShift Bot <notifications@github.com> wrote:

continuous-integration/openshift-jenkins/merge FAILURE (
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_future/409/)
(Base Commit: 90aaad3
<https://github.com/openshift/origin/commit/90aaad331e9b68a3685c04b305e0da905d47570b>
)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/12982#issuecomment-281531053>,
or mute
the thread

**Comment by eparis** [comment]
[merge] is this some sick joke at this point?

**Comment by smarterclayton** [comment]
The merge queue hates this PR.

**Comment by eparis** [comment]
[merge] if this fails, i'm just going to hit the big green button on all the PRs that are killing me.

**Comment by smarterclayton** [comment]
As steves email noted we may have a problem with Jenkins.  If necessary
we'll drain the queue manually.

On Feb 22, 2017, at 1:53 PM, Eric Paris <notifications@github.com> wrote:

[merge] if this fails, i'm just going to hit the big green button on all
the PRs that are killing me.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/12982#issuecomment-281764859>,
or mute
the thread

**Comment by smarterclayton** [comment]
Flake https://github.com/openshift/origin/issues/13068

Merging by hand.

**Pull request merged**